src/display.c

```diff
@@ -479,42 +486,53 @@ void display_close()
     for( int i = 0; i < __buffers; i++ )
     {
         /* Free framebuffer memory */
-        if( buffer[i] )
+        if( __safe_buffer[i] )
         {
-            free( buffer[i]);
+            free_uncached( __safe_buffer[i]);
         }
 
-        buffer[i] = 0;
-        __safe_buffer[i] = 0;
+        __safe_buffer[i] = NULL;
     }
 
+    free(surfaces);
+    surfaces = NULL;
+
     enable_interrupts();
 }
 
 /**
  * @brief Lock a display buffer for rendering
  *
  * Grab a display context that is safe for drawing.  If none is available
```

Comment:
Should we still use the term "display context" in documentation or just switch to "surface"?